astroplan/constraints.py

```diff
@@ -83,6 +84,92 @@ def _get_altaz(times, observer, targets,
     return observer._altaz_cache[aakey]
 
 
+def _get_moon_data(times, observer,
+                   force_zero_pressure=False):
+    """
+    Calculate moon altitude az and illumination for an array of times for ``observer``.
+
+    Cache the result on the ``observer`` object.
+
+    Parameters
+    ----------
+    times : `~astropy.time.Time`
+        Array of times on which to test the constraint
+
+    observer : `~astroplan.Observer`
+        The observer who has constraints ``constraints``
+
+    Returns
+    -------
+    moon_dict : dict
+        Dictionary containing three key-value pairs. (1) 'times' contains the
+        times for the computations, (2) 'altaz' contains the
+        corresponding alt/az coordinates at those times and (3) contains
+        the moon illumination for those times.
+    """
+    if not hasattr(observer, '_moon_cache'):
+        observer._moon_cache = {}
+
+    # convert times to tuple for hashing
+    aakey = (tuple(times.jd))
+
+    if aakey not in observer._moon_cache:
+        try:
+            if force_zero_pressure:
+                observer_old_pressure = observer.pressure
+                observer.pressure = 0
+
+            altaz = observer.moon_altaz(times)
+            illumination = np.array(moon_illumination(times,
+                                                      observer.location))
+            observer._moon_cache[aakey] = dict(times=times,
+                                               illum=illumination,
+                                               altaz=altaz)
+        finally:
+            if force_zero_pressure:
+                observer.pressure = observer_old_pressure
+
+    return observer._moon_cache[aakey]
+
+
+def _get_transit_times(times, observer, targets):
```

Comment:
I think the term "transit" should always appear as "meridian transit", as is the standard in other functions in astroplan. This avoids ambiguity with exoplanet transits.